### PR TITLE
feature: text chunker with overlap

### PR DESCRIPTION
**Pull request by lxobr**
## Description

- Implements `TextChunkerWithOverlap` with configurable `chunk_overlap_ratio` 
- Abstracts chunk_data generation via `get_chunk_data` callable (defaults to `chunk_by_paragraph`)
- Parametrized tests verify `TextChunker` and `TextChunkerWithOverlap` (0% overlap) produce identical output for all edge cases.
- Overlap-specific tests validate `TextChunkerWithOverlap` behavior 

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Other (please specify):

## Pre-submission Checklist

- [x] **I have tested my changes thoroughly before submitting this PR**
- [x] **This PR contains minimal changes necessary to address the issue/feature**
- [x] My code follows the project's coding standards and style guidelines
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if applicable)
- [x] All new and existing tests pass
- [x] I have searched existing PRs to ensure this change hasn't been submitted already
- [ ] I have linked any relevant issues in the description
- [x] My commits have clear and descriptive messages

## DCO Affirmation
I affirm that all code in every commit of this pull request conforms to the terms of the Topoteretes Developer Certificate of Origin.

### CODE REVIEW

File: cognee/modules/chunking/text_chunker_with_overlap.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+from cognee.shared.logging_utils import get_logger
```
Comment (hajdul88) [inline cognee/modules/chunking/text_chunker_with_overlap.py:1]: The naming of the file doesn't match with the other chunkers, since its a class should be TextChunkeWithOverlap

**Review by hajdul88 — COMMENTED** [review]
Tested it, discussed with Lazar

File: cognee/modules/chunking/text_chunker_with_overlap.py (hunk 1 above)
Comment (lxobr) [reply cognee/modules/chunking/text_chunker_with_overlap.py:1]: I agree, that’s a fair point. My hope is that this chunker will eventually replace the current default. The other chunkers aren’t currently used and likely need a closer review. Rather than continuing with the existing naming pattern that violates PEP 8, I’d prefer to keep this file name as is and create a follow-up issue to review and rename the other chunkers for consistency.

**Review by hajdul88 — APPROVED** [review]
Thanks for the quick change, otherwise LGTM
